vnext/Desktop/Modules/WebSocketModule.cpp

```diff
@@ -147,23 +147,36 @@ std::shared_ptr<IWebSocketResource> WebSocketModule::GetOrCreateWebSocket(int64_
   if (itr == m_webSockets.end())
   {
     auto ws = IWebSocketResource::Make(std::move(url));
-    ws->SetOnError([this, id, ws](const IWebSocketResource::Error& err)
+    auto weakInstance = this->getInstance();
+    ws->SetOnError([this, id, ws, instance = weakInstance.lock()](const IWebSocketResource::Error& err)
     {
+      if (!instance)
```

Comment:
nit: since we're passing a shared_ptr already, we can avoid the duplicate null checks if we lock + return early after acquiring the weakInstance.

Reply:
Talked offline. Use of a strong ptr in the capture group here could actually cause reference cycles. We should capture a weak_ptr.